controller/src/main/java/org/jboss/as/controller/persistence/yaml/YamlConfigurationExtension.java

```diff
@@ -0,0 +1,713 @@
+                        attributes.add(def);
+                    }
+                }
+            }
+        }
+        attributes.addAll(Arrays.asList(operationEntry.getOperationDefinition().getParameters()));
+        ModelNode op = createOperation(address, operationEntry);
+        for (AttributeDefinition att : attributes) {
+            if (map.containsKey(att.getName())) {
+                Object value = map.get(att.getName());
+                map.remove(att.getName());
+                switch (att.getType()) {
+                    case OBJECT:
+                        op.get(att.getName()).set(processObjectAttribute((ObjectTypeAttributeDefinition) att, (Map<String, Object>) value));
+                        break;
+                    case LIST:
+                        ModelNode list = op.get(att.getName()).setEmptyList();
+                        processListAttribute((ListAttributeDefinition) att, list, value);
+                        break;
+                    default:
+                        op.get(att.getName()).set(value.toString());
+                        break;
+                }
+            }
+        }
+        ParsedBootOp operation = new ParsedBootOp(op, operationEntry.getOperationHandler());
+        MGMT_OP_LOGGER.debugf("Adding resource with operation %s", op);
+        postExtensionOps.add(operation);
+    }
+
+    private ModelNode createOperation(PathAddress address, OperationEntry operationEntry) {
+        ModelNode op = new ModelNode();
+        op.get(OP).set(operationEntry.getOperationDefinition().getName());
+        op.get(OP_ADDR).set(address.toModelNode());
+        return op;
+    }
+
+    @SuppressWarnings("unchecked")
+    private ModelNode processObjectAttribute(ObjectTypeAttributeDefinition att, Map<String, Object> map) {
+        ModelNode objectNode = new ModelNode();
+        for (AttributeDefinition child : att.getValueTypes()) {
+            if (map.containsKey(child.getName())) {
+                Object value = map.get(child.getName());
+                switch (child.getType()) {
+                    case OBJECT:
+                        objectNode.get(child.getName()).set(processObjectAttribute((ObjectTypeAttributeDefinition) child, (Map<String, Object>) value));
+                        break;
+                    case LIST:
+                        ModelNode list = objectNode.get(child.getName()).setEmptyList();
+                        processListAttribute((ListAttributeDefinition) child, list, value);
+                        break;
+                    default:
+                        objectNode.get(child.getName()).set(value.toString());
+                        break;
+                }
+            }
+        }
+        return objectNode;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void processListAttribute(ListAttributeDefinition att, ModelNode list, Object value) {
+        AttributeDefinition type = att.getValueAttributeDefinition();
```

Comment:
what if this method raises an exception?

Reply:
Then the configuration would fail and the server wouldn't boot until you fix your YAML configuration or the attribute definition. Basically that means that this attribute is not configurable from YAML.

Reply:
I'm splitting hairs here but the only thing you need is to check if the `att` is an instance of `ObjectListAttributeDefinition` or `PrimitiveListAttributeDefinition` or `SimpleListAttributeDefinition` and process accordingly.

Reply:
or even just check if it is an instance of ObjectListAttributeDefinition and handles all other types of list the same way (with a String item)

Reply:
I disagree (I have long hair) as if there is some ListAttributeDefinition in the outer space that wants to manipulate an ObjectAttributeDefinition without extending ObjectListAttributeDefinition this switch over the implementation would 'break'.

Reply:
@jmesnil for what is worth, it seems better to me to fail "loudly" and not boot, so the config is fixed (and the lack of support for the requested features explicit).